src/app/Dashboard/AddCard.tsx

```diff
@@ -157,3 +226,166 @@ export const AddCard: React.FunctionComponent<AddCardProps> = (props: AddCardPro
+            <Switch label="On" labelOff="Off" isChecked={props.config[ctrl.key]} onChange={handleChange(ctrl.key)} />
+          );
+          break;
+        case 'number':
+          input = (
+            <NumberInput
+              inputName={ctrl.name}
+              inputAriaLabel={`${ctrl.name} input`}
+              value={props.config[ctrl.key]}
+              onChange={handleNumeric(ctrl.key)}
+              onPlus={handleNumericStep(ctrl.key, 1)}
+              onMinus={handleNumericStep(ctrl.key, -1)}
+            />
+          );
+          break;
+        case 'string':
+          input = (
+            <TextInput
+              type="text"
+              aria-label={`${ctrl.key} input`}
+              value={props.config[ctrl.key]}
+              onChange={handleChange(ctrl.key)}
+            />
+          );
+          break;
+        case 'text':
+          input = (
+            <TextArea
+              type="text"
+              aria-label={`${ctrl.key} input`}
+              value={props.config[ctrl.key]}
+              onChange={handleChange(ctrl.key)}
+            />
+          );
+          break;
+        case 'select':
+          input = (
+            <SelectControl handleChange={handleChange(ctrl.key)} config={props.config[ctrl.key]} control={ctrl} />
+          );
+          break;
+        default:
+          input = <Text>Bad config</Text>;
+          break;
+      }
+      return (
+        <FormGroup key={`${ctrl.key}}`} label={ctrl.name} helperText={ctrl.description} isInline isStack>
+          {input}
+        </FormGroup>
+      );
+    },
+    [props.config, handleChange, handleNumeric, handleNumericStep]
+  );
+
+  return (
+    <>
+      {props.controls.length > 0 ? (
+        <Form>
+          <Title headingLevel={'h5'}>Configure the {props.cardTitle} card</Title>
+          {props.controls.map((ctrl) => createControl(ctrl))}
+        </Form>
+      ) : (
+        <Text>No configuration required.</Text>
+      )}
+    </>
+  );
+};
+
+const SelectControl = (props: { handleChange: ({}) => void; control: PropControl; config: any }) => {
+  const addSubscription = useSubscriptions();
+
+  const [selectOpen, setSelectOpen] = React.useState(false);
+  const [options, setOptions] = React.useState([] as Element[]);
+
+  const handleSelect = React.useCallback(
+    (_, selection, isPlaceholder) => {
+      if (!isPlaceholder) {
+        props.handleChange(selection);
+      }
+      setSelectOpen(false);
+    },
+    [props.handleChange, setSelectOpen]
+  );
+
+  React.useEffect(() => {
+    let obs;
+    if (props.control.values instanceof Observable) {
+      obs = props.control.values;
+    } else {
+      obs = of(props.control.values);
+    }
+    addSubscription(
+      obs.subscribe({
+        next: (v) =>
+          setOptions((old) => {
+            if (Array.isArray(v)) {
+              return v;
+            }
+            return [...old, v];
+          }),
+        complete: () => {},
```

Comment:
Any need from this callback? Maybe we should handle error instead?

Reply:
No, it's not needed - I'll remove it.

Not sure what to do for error handling. Probably nothing here. I figure this is probably going to be an Observable returned by an `Api.service` method which will already handle the error and display a notification. I guess I could wipe out the existing options and add a fake option that just says `Error` or something?

Reply:
> I guess I could wipe out the existing options and add a fake option that just says Error or something?

Or i guess we could just wipe out and left as empty. Or maybe show some custom compact error view instead of the select component should work too.

Reply:
But probably later since lots of things could change...

Reply:
Oh yeh just saw the new commit. Looking good :D